apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -860,3 +922,20 @@ def _find_extent_for_domain(
         return np.datetime64(iextent, "ns")
 
     return extent
+
+
+def _revise_domain_for_extent(
+    domain: Tuple[Any, Any], extent: Any, saturated_range: bool
+) -> Tuple[Any, Any]:
+    """
+    We need to do this to avoid this error at array-creation time:
+
+    Error: [TileDB::Dimension] Error: Tile extent check failed; domain max
```

Comment:
This is just the error description right? The exception type should be `SOMAError` and not `TileDBError`.